include/dpp/cluster.h

```diff
@@ -436,6 +468,52 @@ class DPP_EXPORT cluster {
 
 	/* Functions for attaching to event handlers */
 
+	/**
+	 * @brief Register a slash command handler.
+	 *
+	 * @param name The name of the slash command to register
+	 * @param handler A handler function of type `slashcommand_handler_t`
+	 *
+	 * @return bool Returns `true` if the command was registered successfully, or `false` if
+	 * the command with the same name already exists
+	 */
+	bool register_command(const std::string& name, const slashcommand_handler_t handler);
+
+#ifdef DPP_CORO
+	/**
+	 * @brief Register a coroutine-based slash command handler.
+	 *
+	 * @param name The name of the slash command to register.
+	 * @param handler A coroutine handler function of type `co_slashcommand_handler_t`.
+	 *
+	 * @return bool Returns `true` if the command was registered successfully, or `false` if
+	 * the command with the same name already exists.
+	 */
+	template <typename F>
+	std::enable_if_t<std::is_same_v<std::invoke_result_t<F, const slashcommand_handler_t&>, dpp::task<void>>, bool>
+	register_command(const std::string& name, F&& handler){
+		std::unique_lock lk(named_commands_mutex);
+		if (named_commands.count(name) != 0) {
+			return false;
+		}
+		named_commands.emplace(name, handler);
```

Comment:
Why not
```cpp
auto [_, inserted] = named_commands.try_emplace(name, handler);
return inserted;
```
Saves the double lookup, and actually, should std::forward handler here too